### PR TITLE
Expose training script

### PR DESCRIPTION
**Pull request by adhardydm**
First step to supporting custom units/entities/models: 
- expose the training script to the command line when a user installs the package.
- add a [classifier] extras so users can more easily install the classifier dependencies
- add support for custom training data and custom save file location in the training script

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ script:
   # Note that the classifier trained here is included in the target package
   - coverage run -a --source=quantulum3 setup.py test -s quantulum3.tests.test_classifier.ClassifierBuild
   - coverage run -a --source=quantulum3 setup.py test -s quantulum3.tests.test_classifier.ClassifierTest
+  - coverage run -a --source=quantulum3 setup.py test -s quantulum3.tests.test_scripts.TrainScriptTest
```
Comment (nielstron) [inline .travis.yml:50]: Does this generate a new classifier that overwrites the initially trained one?

**Comment by nielstron** [comment]
I am assuming that this does not yet include the option to also use this classifier in place of the original classifier?

**Comment by adhardydm** [comment]
Not yet - will roll that in next when I look at loading custom unit and entity files.

File: .travis.yml (hunk 1 above)
Comment (adhardydm) [reply .travis.yml:50]: No, I've patched out the actual calls to train_classifier when testing the train script, so that never actually runs.

**Pull request merged**